Dockerfile.development

```diff
@@ -11,9 +10,12 @@ WORKDIR /gossamer
 
 COPY . .
 
-RUN go get ./...
-RUN go install -trimpath github.com/ChainSafe/gossamer/cmd/gossamer
+ARG GO_BUILD_FLAGS
+RUN go build \
+    -trimpath \
+    -o /gossamer/bin/gossamer \
+    ${GO_BUILD_FLAGS} \
+    ./cmd/gossamer
 
-RUN ["sh", "-c", "gossamer init --chain=${chain} --base-path=${basepath}/${chain}"]
```

Comment:
This change will have implications for our infrastructure setup. 
- For the metal infrastructure, it will require implementing a small task to initialize the node as part of the deployment process. 
- For the ECS-based setup, adjustments to the task definition will be necessary. This includes adding a new container definition for the initialization step and defining a dependency to ensure the main run container starts only after the initialization process completes successfully.

Reply:
Sorry, then I misunderstood something on our last call. I thought `init` was already run during the deployment process. I don't see how running it during build time would work though. Whatever files gossamer writes to `basepath/...` during `init` won't be there during runtime because a volume has been mounted to this location. 🤔

I guess it's time to look into what `init` actually does and whether it's needed...

Reply:
@Aminechakr I've double checked and `init` is not required to start a fresh node, so this line can be safely removed.